### DISCUSSION TITLE
Upload file during plugin entry creation

### DISCUSSION THREAD

**Discussion by Fareroo7** [strapi/strapi#7090] [Legacy Topics]
Hello,

I followed the docs to handle file upload during entity creation. ([Link](https://strapi.io/documentation/v3.x/plugins/upload.html#upload-file-during-entry-creation ))

With the help of this documentation, I create the service to handle the plugin's model: [Link](https://strapi.io/documentation/v3.x/concepts/services.html#extending-a-model-service)

In the documentation you can clearly see: 

> if you are using a plugin's model you will have to add the `plugin` key (plugin: 'users-permissions')

but it didn't work.

After some debugging, I saw that the key  `plugin` was never used.
In the code, the key for specifying the plugin is called `source`.

With the key `source` everything is working great.

Not sure if there is just a documentation updated needed or the source code needs an update.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by derrickmehaffy** [comment]
Thank you for the information @Fareroo7 I'll add this to my to-do list to update the documentation, you are correct the field should be `source` not `plugin`
